lib/openhab/dsl/monkey_patch/items/persistence.rb

```diff
@@ -31,35 +47,48 @@ def previous_state(service = nil, skip_equal: false)
             PersistenceExtensions.previous_state(self, skip_equal, service&.to_s)
           end
 
-          %w[
-            average_since
-            changed_since
-            delta_since
-            deviation_since
-            evolution_rate
-            historic_state
-            maximum_since
-            minimum_since
-            sum_since
-            updated_since
-            variance_since
-          ].each do |method|
+          PERSISTENCE_METHODS.each do |method|
             define_method(method) do |timestamp, service = nil|
               service ||= persistence_service
-              if timestamp.is_a? Java::JavaTimeTemporal::TemporalAmount
-                timestamp = Java::JavaTime::ZonedDateTime.now.minus(timestamp)
-              end
-              result = PersistenceExtensions.public_send(method, self, timestamp, service&.to_s)
-              if result.is_a?(Java::OrgOpenhabCoreLibraryTypes::DecimalType) && respond_to?(:unit) && unit
-                Quantity.new(Java::OrgOpenhabCoreLibraryTypes::QuantityType.new(result.to_big_decimal, unit))
-              else
-                result
-              end
+              result = PersistenceExtensions.public_send(method, self, to_zdt(timestamp), service&.to_s)
+              QUANTITY_METHODS.include?(method) ? quantify(result) : result
             end
           end
 
           private
 
+          #
+          # Convert timestamp to ZonedDateTime if it's a TemporalAmount
+          #
+          # @param [Object] timestamp to convert
+          #
+          # @return [ZonedDateTime]
+          #
+          def to_zdt(timestamp)
+            if timestamp.is_a? Java::JavaTimeTemporal::TemporalAmount
+              logger.trace("Converting #{timestamp} (#{timestamp.class}) to ZonedDateTime")
+              Java::JavaTime::ZonedDateTime.now.minus(timestamp)
+            else
+              timestamp
+            end
+          end
+
+          #
+          # Convert value to Quantity if it is a DecimalType and a unit is defined
+          #
+          # @param [Object] value The value to convert
+          #
+          # @return [Object] Quantity or the original value
+          #
+          def quantify(value)
```

Comment:
That is a better method name than we use elsewhere.. We should convert the others I will open an issue.